### PR TITLE
Refactor candidate view, fix 404's for future candidates

### LINKED ISSUE #2294: 404 Errors when clicking through to individual candidate/committee pages
Some candidates and committees that show up in the initial datatable results for `data/candidates` and `data/committees` get 404 errors when you click through to view their individual pages:
Examples:
Candidates:
MENEFEE, JON MARK MR : https://www.fec.gov/data/candidate/H0TX09168/ 

MARTIN, JOSEPH S: https://www.fec.gov/data/candidate/P40003832/

~Committees~ Fixed by PR #2296 
SIERRA CLUB: https://www.fec.gov/data/committee/C90011875/

### PR DESCRIPTION
**Pull request by lbeaufort**
## Summary (required)

- Replaces PR https://github.com/fecgov/fec-cms/pull/2316 - bad branch name
- Resolves #2294 
Refactor candidate view, fix 404's for future cycle candidates without committees, simplify 2-year select macro and use it for the candidate raising/spending pages

## Impacted areas of the application
List general components of the application that this PR will affect:

- Candidate profile pages
- Candidate raising/spending pages now show two-year period label (that's what we display) instead of election (election label was incorrect). Committee two-year period dropdowns should be unchanged - renamed to `two_year_select`

- Test previously 404-ing candidate pages: 
http://localhost:8000/data/candidate/H0TX09168/
http://localhost:8000/data/candidate/P40003832/
- Test future/past House/Senate/Presidential pages

## Before
screen shot 2018-08-29 at 9 43 12 am

## After
screen shot 2018-08-29 at 9 43 21 am

## Related PRs
#2302 *Add tests for candidate view*
#2311  *Refactor tests for candidate view*

### CODE REVIEW

File: fec/data/templates/macros/cycle-select.jinja [hunk 1]
```diff
@@ -27,27 +27,12 @@
 {% endif %}
 {% endmacro %}
 
-{% macro candidate_cycle_select(cycles, cycle=none, id='') %}
-{% set cycle = cycle | int %}
-  <div class="content__section">
-    <label for="{{id}}-cycle" class="label cycle-select__label">Election</label>
-    <select id="{{id}}-cycle" class="js-cycle" name="cycle" data-cycle-location="query" data-election-full="False">
```
Comment (lbeaufort) [inline fec/data/templates/macros/cycle-select.jinja]: This should be `data-election-full="false", lower case (fixed with this commit). It was causing a bug where going from raising/spending pages on an off-election year and then back to the financial summary tab wasn't selecting the 2-year time period button:
screen shot 2018-08-29 at 4 49 23 pm

File: fec/data/templates/partials/candidate/about-candidate.jinja [hunk 2]
```diff
@@ -48,7 +48,7 @@
                 {% if statement.fec_file_id %}
                   <div class="t-small u-small-icon-padding--left"> {{ statement.fec_file_id }}</div>
                 {% endif %}
-                <div class="u-small-icon-padding--left"> Filed {{ statement.receipt_date }}</div>
+                <div class="u-small-icon-padding--left"> Filed {{ statement.receipt_date|date }}</div>
```
Comment (vrajmohan) [inline fec/data/templates/partials/candidate/about-candidate.jinja]: Nice!

File: fec/data/views.py [hunk 3]
```diff
@@ -126,7 +126,7 @@ def get_candidate(candidate_id, cycle, election_full):
 
     result_type = 'candidates'
     duration = election_durations.get(candidate['office'], 2)
-    min_cycle = cycle - duration if election_full else cycle
+    cycle_start_year = cycle - duration if election_full else cycle
```
Comment (vrajmohan) [inline fec/data/views.py]: This is still confusing to me. If the cycle is 2016 and the office is "President", isn't the cycle start year 2011 and not 2010?

File: fec/data/views.py [hunk 4]
```diff
@@ -123,97 +136,86 @@ def get_candidate(candidate_id, cycle, election_full):
...
-    # and when calling the API for totals, set election_full to False.
-    # The max_cycle value is also referenced in the templates for setting
-    # the cycle for itemized tables. Because these are only in 2-year chunks,
+    # Because the API only has totals through the current cycle,
     # the cycle should never be beyond the one we're in.
-    cycles = [cycle for cycle in candidate['cycles'] if cycle <= utils.current_cycle()]
-    max_cycle = cycle if cycle <= utils.current_cycle() else utils.current_cycle()
-    show_full_election = election_full if cycle <= utils.current_cycle() else False
+    cycles = [
+        cycle_tmp for cycle_tmp in candidate['cycles']
```
Comment (vrajmohan) [inline fec/data/views.py]: Minor - it is cleaner to use `cycle` as the list comprehension variable as done in line 164 below.

File: fec/data/views.py [hunk 5]
```diff
@@ -252,8 +254,7 @@ def candidate(request, candidate_id):
     if cycle is not None:
         cycle = int(cycle)
 
-    election_full = request.GET.get('election_full', True)
-    election_full = bool(strtobool(str(election_full)))
+    election_full = bool(strtobool(request.GET.get('election_full', 'True')))
```
Comment (vrajmohan) [inline fec/data/views.py]: Minor - do we really need `distutils.util.strtobool`? Isn't the input controlled by us or specified in our API?

File: fec/data/views.py [hunk 6]
```diff
@@ -105,32 +105,6 @@ def get_candidate(candidate_id, cycle, election_full):
         election_full=show_full_election,
     )
 
-    # cycle corresponds to the two-year period for which the committee has
```
Comment (vrajmohan) [inline fec/data/views.py]: Well done! Nice work on the commit message as well.

File: fec/data/views.py [hunk 7]
```diff
@@ -127,20 +127,6 @@ def get_candidate(candidate_id, cycle, election_full):
         if cycle_in_range(cycle_tmp)
     ]
 
-    # Annotate committees with most recent available cycle
```
Comment (vrajmohan) [inline fec/data/views.py]: The commit message could say more about why this complex calculation was not needed, other than the fact that `max` was throwing errors.

File: fec/data/views.py (hunk 5 above)
Comment (lbeaufort) [reply fec/data/views.py]: I agree it seems overly complex - I'm planning to address this [here](https://github.com/fecgov/fec-cms/issues/2326) so I can test thoroughly.

**Pull request merged**